src/styles/_catalog.scss

```diff
@@ -1,22 +1,28 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  justify-content: center;
+  gap: 48px;
 
-  max-width: 944px;
-  margin: 0 auto;
+  grid-template-columns: $card-width;
 
   &__card {
-    display: flex;
+    display: grid;
     justify-content: center;
     align-items: center;
-    width: 200px;
     height: 300px;
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
   }
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+  @media(min-width: 488px) {
+    grid-template-columns: repeat(2, $card-width);
+  }
+
+  @media(min-width: 768px) {
+    grid-template-columns: repeat(3, $card-width);
+  }
+
+  @media(min-width: 1024px) {
+    grid-template-columns: repeat(4, $card-width);
   }
```

Comment:
You can make a loop to create all media queries. Just create a correctly mapped list to iterate